### PR TITLE
Add pytest test suite with coverage and CI workflow

### PR DESCRIPTION
**Pull request by TheRealAgentK**
## Summary

Adds a comprehensive pytest-based test suite (72 tests, 99% coverage) and GitHub Actions CI workflow.

### Test suite
- **`tests/test_results.py`** (14 tests) — dataclasses (`ActionResult`, `ActionError`, `ConnectedAccountInfo`, `IntegrationResult`, `FetchResponse`), exceptions, and enums
- **`tests/test_integration.py`** (36 tests) — config loading, handler registration, action/trigger/connected-account execution, auth validation, interval parsing, edge cases
- **`tests/test_execution_context.py`** (22 tests) — HTTP methods, error handling, auth injection, query params, retries, form encoding, timeouts, JSON parse fallback, context manager lifecycle

### CI/CD
- GitHub Actions workflow (`.github/workflows/tests.yml`) runs on push to `master` and PRs
- Coverage report via `pytest-cov` with `MishaKav/pytest-coverage-comment` posting results to PRs
- `set -o pipefail` ensures test failures correctly fail the CI step

### Documentation
- **`AGENTS.md`** — testing guidelines for AI agents
- **`README.md`** — testing section for developers
- **`RELEASING.md`** — run tests before building/uploading

### Dependencies
Added `[project.optional-dependencies] test` with `pytest`, `pytest-asyncio`, `aioresponses`, `pytest-cov`

### Coverage
The only 3 uncovered lines are abstract `pass` bodies in `ActionHandler`, `PollingTriggerHandler`, and `ConnectedAccountHandler` — untestable by design.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          python -m pip install --upgrade pip
+          pip install -e ".[test]"
+
+      - name: Run tests with coverage
+        run: |
+          python -m pytest tests/ -v --tb=short \
+            --cov=autohive_integrations_sdk \
+            --cov-report=term-missing \
+            --cov-report=xml:coverage.xml \
+            | tee pytest-coverage.txt
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply .github/workflows/tests.yml:54]: Codex was right, fixed.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Tests
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
[the remark this answers is not in the document]
Comment (TheRealAgentK) [reply .github/workflows/tests.yml:7]: Incorrect comment.

**Review by ProRedCat — APPROVED** [review]

**Review by ProRedCat — APPROVED** [review]

**Comment by TheRealAgentK** [comment]
Thx @ProRedCat - added a few more tweaks around the coverage output, but that doesn't make a material difference to the core of the changes. Merged.
